pydp/algorithms/_algorithm.py

```diff
@@ -21,14 +26,21 @@ def __init__(self, **kwargs):
         self._l0_sensitivity = kwargs.get("l0_sensitivity", "Not set")
         self._linf_sensitivity = kwargs.get("linf_sensitivity", "Not set")
 
+    @staticmethod
+    def __check_input(name: str, value: float):
+        if math.isnan(value) or math.isinf(value):
+            raise ValueError(
+                "invalid value '{}' for paramater '{}'.".format(value, name)
+            )
+
     @staticmethod
     def __map_dtype_str(dtype: str):
         if dtype == "int":
             return "Int"
         elif dtype == "float":
             return "Double"
         else:
-            raise RuntimeError("dtype: {} is not supported".format(dtype))
+            raise ValueError("dtype '{}' is not supported.".format(dtype))
```

Comment:
Shouldn't this be TypeError?

Reply:
TypeError is used when an unexpected Python type is received. In this case the type of "dtype" is str and we do not accept values outside the main cases ("int" and "float" for the moment).